test/test-auth.js

```diff
@@ -0,0 +1,208 @@
+            firstName,
+            lastName
+          });
+        })
+    });
+  });
+
+  describe('/api/auth/refresh', function() {
+    it('Should reject requests with no credentials', function() {
+      return chai.request(app)
+        .post('/api/auth/refresh')
+        .then(() => expect.fail(null, null, 'Request should not succeed'))
+        .catch(err => {
+          if (err instanceof chai.AssertionError) {
+            throw err;
+          }
+
+          const res = err.response;
+          expect(res).to.have.status(401);
+        });
+    });
+    it('Should reject requests with an invalid token', function() {
+      const token = jwt.sign({
+        username,
+        firstName,
+        lastName
+      }, 'wrongSecret', {
+        algorithm: 'HS256',
+        expiresIn: '7d'
+      });
+
+      return chai.request(app)
+        .post('/api/auth/refresh')
+        .set('Authorization', `Bearer ${token}`)
+        .then(() => expect.fail(null, null, 'Request should not succeed'))
+        .catch(err => {
+          if (err instanceof chai.AssertionError) {
+            throw err;
+          }
+
+          const res = err.response;
+          expect(res).to.have.status(401);
+        });
+    });
+    it('Should reject requests with an expired token', function() {
+      const token = jwt.sign({
+        user: {
+          username,
+          firstName,
+          lastName
+        },
+        exp: Math.floor(Date.now() / 1000) - 10 // Expired ten seconds ago
+      }, JWT_SECRET, {
+        algorithm: 'HS256',
+        subject: username
+      });
+
+      return chai.request(app)
+        .post('/api/auth/refresh')
+        .set('authorization', `Bearer ${token}`)
+        .then(() => expect.fail(null, null, 'Request should not succeed'))
+        .catch(err => {
+          if (err instanceof chai.AssertionError) {
+            throw err;
+          }
+
+          const res = err.response;
+          expect(res).to.have.status(401);
+        });
+    });
+    it('Should return a valid auth token with a newer expiry date', function() {
+      const token = jwt.sign({
+        user: {
+          username,
+          firstName,
+          lastName
+        },
+      }, JWT_SECRET, {
+        algorithm: 'HS256',
+        subject: username,
+        expiresIn: '7d'
+      });
+      const decoded = jwt.decode(token);
+
+      return chai.request(app)
+        .post('/api/auth/refresh')
+        .set('authorization', `Bearer ${token}`)
+        .then(res => {
+          expect(res).to.have.status(200);
+          expect(res.body).to.be.an('object');
+          const token = res.body.authToken;
+          expect(token).to.be.a('string');
+          const payload = jwt.verify(token, JWT_SECRET, {
+            algorithm:  ["HS256"]
+          });
+          expect(payload.user).to.deep.equal({
+            username,
+            firstName,
+            lastName
+          });
+          expect(payload.exp).to.be.at.least(decoded.exp);
```

Comment:
This is not a guarantee that expiry is later. if same expiry, this assertion still passes.